source/VersionHandlerImpl/src/ProjectSettings.cs

```diff
@@ -234,8 +253,14 @@ internal class InMemorySettings : ISettings {
         /// </summary>
         /// <param name="name">Name of the value.</param>
         /// <param name="value">Value to set.</param>
-        private void Set<T>(string name, T value) {
-            settings[name] = value.ToString();
+        private void Set<T>(string name, T value)
+        {
+            string stringValue = value.ToString();
+
+            if (!settings.ContainsKey(name) || settings[name] != stringValue) 
```

Comment:
To match the current coding style,readability, and portability, this if statement should have enclosing curly braces around the set call for IsModified.